### ISSUE TITLE
Unused type alias does not produce warning

### ISSUE THREAD

**Issue by matthew-piziak** [rust-lang/rust#37455]
I created the following `main.rs`:

``` rust
type Unused = String;

fn main() {
    println!("Hello, world!");
}
```

Note that `Unused` is not `pub` and not used. Nevertheless no "unused" warning is given.

Here is the full (empty) output of `cargo build`:

``` shell
Finished debug [unoptimized + debuginfo] target(s) in 0.0 secs
```
